### ISSUE TITLE
command 'perforce.Refresh' not found and similar errors.

### ISSUE THREAD

**Issue by JimmyDeemo** [stef-levesque/vscode-perforce#120]
I have found that in the latest version of VSCode some of the SCM interface functions do not perform as expected. Notably if I was to follow the gif in the extension details window within VS Code I get the error `command 'perforce.Refresh' not found`.

I understand that many have had issues actually getting P4 setup within VS Code, after searching the issues. However I appear to have connected fine. If I click the Perforce text in the bottom left of the VS Code window I can use all of the short commands in that list. However anything in the main command palette, prefixed with `Perforce:` doesn't appear to work.

e.g. Trying to display client information yields the `command 'perforce.info' not found` error.

VSCode Version: 1.19.2
Perforce for VS Code version: 3.1.0

**Comment by stef-levesque** [comment]
Please try the following:
1. `View: Toggle Output`
1. Look for `Perforce Log` in the right-side dropdown
1. See if there is any error message

Also, you can try to run `p4 info` from the integrated terminal, to see if your `.p4config` is properly setup.

**Comment by JimmyDeemo** [comment]
I was able to view the perforce log fine, here is what was output on a newly opened project (no errors as far as I can see, please excuse redacted edits in '[]');

```
p4 -u imac -c iMac_iMacOne -p [MY_URL]:1666 -P [MY_PASS] info
p4 -u imac -c iMac_iMacOne -p [MY_URL]:1666 -P [MY_PASS] login -s
p4 -u imac -c iMac_iMacOne -p [MY_URL]:1666 -P [MY_PASS] set -q
p4 -u imac -c iMac_iMacOne -p [MY_URL]:1666 -P [MY_PASS] info
p4 -u imac -c iMac_iMacOne -p [MY_URL]:1666 -P [MY_PASS] login -s
p4 -u imac -c iMac_iMacOne -p [MY_URL]:1666 -P [MY_PASS] changes -c iMac_iMacOne -s pending
p4 -u imac -c iMac_iMacOne -p [MY_URL]:1666 -P [MY_PASS] describe -Ss 12450
p4 -u imac -c iMac_iMacOne -p [MY_URL]:1666 -P [MY_PASS] describe -Ss 10684
p4 -u imac -c iMac_iMacOne -p [MY_URL]:1666 -P [MY_PASS] opened
p4 -u imac -c iMac_iMacOne -p [MY_URL]:1666 -P [MY_PASS] fstat “//[SOME_VALID_PATH]/DebugLogger.cs"
p4 -u imac -c iMac_iMacOne -p [MY_URL]:1666 -P [MY_PASS] opened "[SOME_VALID_PATH]/AmazonAccess.cs"
p4 -u imac -c iMac_iMacOne -p [MY_URL]:1666 -P [MY_PASS] opened "[SOME_VALID_PATH]/AmazonAccess.cs"
p4 -u imac -c iMac_iMacOne -p [MY_URL]:1666 -P [MY_PASS] opened "[SOME_VALID_PATH]/AmazonAccess.cs"
```

Results of the info command are as follows (everything looks good, excuse redaction again);
```
User name: imac
Client name: iMac_iMacOne
Client host: iMacs-iMac.local
Client root: /Users/imacone/P4
Current directory: /Users/imacone/P4/ProjectName
Peer address: [VALID_IP]
Client address: [VALID_IP]
Server address: [VALID_IP]
Server root: /mnt/box2/perforce
Server date: 2018/01/24 11:20:28 +0000 GMT
Server uptime: 581:48:40
Server version: P4D/LINUX26X86_64/2017.2/1594901 (2017/11/29)
ServerID: main
Server services: standard
Server license: [VALID_LICENCE]
Server license-ip: [VALID_IP]
Case Handling: insensitive
```

Interestingly an attempt at doing `>Perforce: Show Output` command in VSCode produces a similar 'not found' error. It's as if the extension isn't hooked up to VSCode anymore or the hooks have changed. May I ask, the commands that I get access to by clicking the 'Perforce' text in the blue bar at the bottom of the UI, what do I do to get those commands in the Command Palette?

**Comment by Gerkinfeltser** [comment]
I'm getting similar issues as Jimmy. Commands added by the extension show up in the Command Palette but running any of them results in the `command 'perforce.whatevs' not found`. Unlike Jimmy, I have no entry `Perforce Log` in the output console and the Source Control tab of the side bar claims "There are no active source control providers." Is something else amiss? Thanks for your time!

**Comment by JimmyDeemo** [comment]
I think that the latter part of your problem is due to P4 not being setup properly. I suggest you check your vscode settings and fill out any of the values you can; the equivolent of setting the P4PORT, P4USER, P4CLIENT etc. in terminal.

**Comment by stef-levesque** [comment]
@Gerkinfeltser , as Jimmy said, if you don't see the P4 icon in the status bar, then the Extension haven't loaded. You should try to configure your workspace as specified in the [Activation section](https://github.com/stef-levesque/vscode-perforce#activation)

**Comment by stef-levesque** [comment]
@JimmyDeemo your issue is really bugging me. The commands used by the P4 icons are the same as the Command Prompt.

Could you try to run the Palette Command `Extensions: Open Extensions Folder`
and see if you have multiple instances of `slevesque.perforce-xxx` ?

**Comment by JimmyDeemo** [comment]
I didn't have multiple instances for 'slevesque.perforce' but I did have 'bam.vscode-perforce-1.1.3' which was, I believe another perforce plugin.

I then realised that this was still active in the extensions list. I removed it and restarted and now the perforce plugin works fine.

So my apologies it looks like this issue was a conflict between the plugins. Thank you very much for your help, hope I didn't take up too much of your time!

**Comment by cswiedler** [comment]
I get this error when I try to edit a file which lives underneath a p4config when I'm not using a workspace. For example, if I right-click on a file in WIndows Explorer and select Edit with Code, it appears that perforce operations then happen with a working dir of my home folder (presumably because there's no workspace). If there's no workspace, perhaps perforce operations could happen in the context of whatever file is currently open in the editor?

**Comment by stef-levesque** [comment]
@cswiedler You are right, the SCM plugin doesn't play well when no workspace is opened. 😞 

If you are often using the same P4 config, you can try to set the default values in you User Settings.

**Comment by JonathanMcDevitt** [comment]
Hi all,

I recently installed the Perforce plugin for VSCode and I'm running into that issue of 'command "perforce.insert_command_here" not found'. I have also consulted the below issues:

https://github.com/stef-levesque/vscode-perforce/issues/96
https://github.com/stef-levesque/vscode-perforce/issues/136

Unfortunately none of the potential workarounds in those issues have worked. I have a `.p4config` file at the root of my VSCode workspace with the various P4 variables specified, and the activation is set to 'always'. The problem is that VSCode doesn't even seem to be able to execute the command; I'm not able to run any 'perforce.insert_here' commands. Unlike @JimmyDeemo I don't have any other P4 plugins for VSCode, nor any form of version/source/revision control plugin.

Does anybody have any ideas?

VSCode version: 1.24.1 (x64)
Perforce for VSCode version: 3.1.1
